container/go/cmd/digester/digester.go

```diff
@@ -0,0 +1,119 @@
+	format = flag.String("format", "", "The format of the source image, (oci, legacy, or docker). The docker format should be a tarball of the image as generated by docker save.")
+	layers utils.ArrayFlags
+)
+
+const (
+	// manifestFile is the filename of image manifest.
+	manifestFile = "manifest.json"
+	// configFile is the filename of image config.
+	configFile = "config.json"
+	// indexManifestFile is the filename of image manifest config in OCI format.
+	indexManifestFile = "index.json"
+)
+
+func main() {
+	flag.Var(&layers, "layers", "The list of paths to the layers of this docker image, only used for legacy images.")
+	flag.Parse()
+	log.Println("Running the Image Digester to generate an image digest file...")
+
+	if *dst == "" {
+		log.Fatalln("Required option -dst was not specified.")
+	}
+	if *src == "" {
+		log.Fatalln("Required option -src was not specified.")
+	}
+	if *format == "" {
+		log.Fatalln("Required option -format was not specified.")
+	}
+
+	// Validates provided format and src path. Check if src is a tarball when pushing a docker image. Trim basename index.json or manifest.json if src is a directory, since we are pushing a OCI/legacy index.
+	var imgSrc string
+	if *format == "docker" && filepath.Ext(*src) != ".tar" {
+		log.Fatalf("Invalid value for argument -src for -format=docker, got %q, want path to tarball file with extension .tar.", *src)
+	}
+	if *format == "legacy" && filepath.Base(*src) != configFile {
+		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to %s", *src, configFile)
+	}
+	if *format == "oci" && filepath.Base(*src) != indexManifestFile {
+		log.Fatalf("Invalid value for argument -src for -format=oci, got %q, want path to %s", *src, indexManifestFile)
+	}
+	if *format == "oci" || *format == "legacy" {
+		imgSrc = filepath.Dir(*src)
+		log.Printf("Determined image source path to be %s based on -format=%s, -src=%s.", imgSrc, *format, *src)
+	}
+	if *format == "docker" {
+		imgSrc = *src
+	}
+	if *format == "legacy" {
+		manifestPath := filepath.Join(imgSrc, manifestFile)
+
+		// TODO (xiaohegong): remove generate manifest after createImageConfig is merged.
+		log.Printf("Generating image manifest to %s...", manifestPath)
+		_, err := compat.GenerateManifest(imgSrc, manifestPath, filepath.Join(imgSrc, configFile), layers)
+		if err != nil {
+			log.Fatalf("Error generating %s from %s: %v", manifestFile, imgSrc, err)
+		}
+	}
+
+	img, err := utils.ReadImage(imgSrc, *format)
+	if err != nil {
+		log.Fatalf("Error reading from %s: %v", imgSrc, err)
+	}
+
+	digest, err := img.Digest()
+	if err != nil {
+		log.Fatalf("Error getting image digest: %v", err)
+	}
+
+	err = WriteDigest(digest, *dst)
+	if err != nil {
+		log.Fatalf("Error outputting digest file to %s: %v", *dst, err)
+	}
+
+	log.Printf("Successfully generated image digest file at %s", *dst)
+}
+
+// WriteDigest outputs digest to a "digest file" at dst. Digest is the image manifest's sha256 hash.
```

Comment:
`WriteDigest` -> `writeDigest`. This function is not being used in any other package.